src/slices/auth.ts

```diff
@@ -0,0 +1,106 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import {
+  AuthUser,
+  fetchAuthSession,
+  fetchUserAttributes,
+  getCurrentUser,
+  signOut,
+} from "aws-amplify/auth";
+
+type User = null | "loading" | AuthenticatedUser;
+
+/** credentials that can be checked before protected page is rendered*/
+export type PreRenderCredential = "ap";
+/** credential that can only be verified once protected page is rendered
+ *  e.g. only after rendering admin:1 can check if user has credential to access admin:1
+ */
+export type PostRenderCredential = number; //EndowmentID
+type Credential = PostRenderCredential | PreRenderCredential;
+
+export type AuthenticatedUser = {
+  token: string;
+  credentials: Credential[];
+  id: string; //email or name or username
+  isSigningOut: boolean;
+};
+
+export const userIsSignedIn = (user: User): user is AuthenticatedUser =>
+  !!(user as AuthenticatedUser).token;
+
+type State = {
+  user: User;
+};
+
+const initialState: State = {
+  user: null,
+};
+
+export const logout = createAsyncThunk("auth/logout", async () => {
+  await signOut();
+});
+
+export const loadSession = createAsyncThunk<User, AuthUser | undefined>(
+  "auth/loadSession",
+  async (user) => {
+    try {
+      const session = await fetchAuthSession();
+      if (!session.tokens) return null;
+
+      const [attributes, _user] = await Promise.all([
+        fetchUserAttributes(),
+        user ? Promise.resolve(user) : getCurrentUser(),
+      ]);
+
+      //remove standard claims
+      const payload = session.tokens.accessToken.payload;
+      const token = session.tokens.accessToken.toString();
+
+      const credentials = [
+        Array.isArray(payload["cognito:groups"]) &&
+        payload["cognito:groups"].includes("AngelProtocolAdmin")
+          ? ("ap" as Credential)
+          : null,
+      ].filter((item): item is Credential => item !== null);
+
+      return {
+        token,
+        credentials,
+        id: attributes.email || _user.username,
+        isSigningOut: false,
+      };
+    } catch (err) {
+      return null;
+    }
+  }
+);
+
+const auth = createSlice({
```

Comment:
ditch `useAuthenticator` hook for user state management due to limitations
* in `v5` retrieval of claims from jwt is of node style instead of react
```
getSession(callback: ((error: Error, session: null) => void) | ((error: null, session: CognitoUserSession) => void), options?: GetSessionOptions | undefined): void
```
* in `v6`, need to call `fetchAuthSession` separately 


instead created `auth` slice which centrally manages user state (updated via auth events)